diff --git a/pkg/c3/portable.h b/pkg/c3/portable.h
index 002276834b..52ebb7817d 100644
--- a/pkg/c3/portable.h
+++ b/pkg/c3/portable.h
@@ -36,7 +36,6 @@
 #     include <sys/time.h>
 #     include <sys/resource.h>
 #     include <sys/mman.h>
-#     include <sys/sendfile.h>
 
 #   elif defined(U3_OS_osx)
 #     include <ctype.h>
diff --git a/pkg/noun/jets.c b/pkg/noun/jets.c
index 45e7e8144a..0003896370 100644
--- a/pkg/noun/jets.c
+++ b/pkg/noun/jets.c
@@ -14,7 +14,7 @@
 #include "retrieve.h"
 #include "serial.h"
 #include "trace.h"
-#include "urcrypt.h"
+#include "sha256.h"
 #include "vortex.h"
 #include "xtract.h"
 
@@ -128,7 +128,7 @@ _cj_bash(u3_noun bat)
       //
       c3_y*    fat_y = sab_u.buf_y;
       c3_y     dig_y[32];
-      urcrypt_shay(fat_y, met_w, dig_y);
+      sha256_easy_hash(fat_y, met_w, dig_y);
 
       pro = u3i_bytes(32, dig_y);
       u3h_put(u3R->jed.bas_p, bat, u3k(pro));
diff --git a/pkg/noun/jets/tree.c b/pkg/noun/jets/tree.c
index 9e920becda..e6e17261f3 100644
--- a/pkg/noun/jets/tree.c
+++ b/pkg/noun/jets/tree.c
@@ -53,7 +53,6 @@ static c3_c* _140_hex_mimes_ha[] = {
   0
 };
 
-    static u3j_harm _140_hex_aes_ecba_en_a[] = {{".2", u3wea_ecba_en}, {}};
     static c3_c* _140_hex_aes_ecba_en_ha[] = {
       "d7674ad72666a787580c52785c5d4d37ca462ba05e904efbeded5d1bd8b02b4b",
       0
@@ -64,16 +63,12 @@ static c3_c* _140_hex_mimes_ha[] = {
       0
     };
   static u3j_core _140_hex_aes_ecba_d[] =
-    { { "en", 7, _140_hex_aes_ecba_en_a, 0, _140_hex_aes_ecba_en_ha },
-      { "de", 7, _140_hex_aes_ecba_de_a, 0, _140_hex_aes_ecba_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_ecba_ha[] = {
     "693409d27b777f73ce92cebd38e9ebceebe1e9c27ad8c9de9afc091e31bd7d9f",
     0
   };
-
-    static u3j_harm _140_hex_aes_ecbb_en_a[] = {{".2", u3wea_ecbb_en}, {}};
     static c3_c* _140_hex_aes_ecbb_en_ha[] = {
       "2c0e3c8f4d741b37324563ecd0ab8fbf87721d1e017f1eeeaf8b6a60515c483b",
       0
@@ -84,149 +79,119 @@ static c3_c* _140_hex_mimes_ha[] = {
       0
     };
   static u3j_core _140_hex_aes_ecbb_d[] =
-    { { "en", 7, _140_hex_aes_ecbb_en_a, 0, _140_hex_aes_ecbb_en_ha },
-      { "de", 7, _140_hex_aes_ecbb_de_a, 0, _140_hex_aes_ecbb_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_ecbb_ha[] = {
     "7255c39f10e068007d0d64dd40e4f6e83492ed2e3222919440be0d28fd42a5e3",
     0
   };
 
-    static u3j_harm _140_hex_aes_ecbc_en_a[] = {{".2", u3wea_ecbc_en}, {}};
     static c3_c* _140_hex_aes_ecbc_en_ha[] = {
       "f56450f2662082e27ba1aecd2fe04c66aa8641d6eb155f8d3707e242a1e1cf1c",
       0
     };
-    static u3j_harm _140_hex_aes_ecbc_de_a[] = {{".2", u3wea_ecbc_de}, {}};
     static c3_c* _140_hex_aes_ecbc_de_ha[] = {
       "28c8c44002799fbe94d4aa07922f2b74dbbf234c84684cd8c3187622b4be6a5f",
       0
     };
   static u3j_core _140_hex_aes_ecbc_d[] =
-    { { "en", 7, _140_hex_aes_ecbc_en_a, 0, _140_hex_aes_ecbc_en_ha },
-      { "de", 7, _140_hex_aes_ecbc_de_a, 0, _140_hex_aes_ecbc_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_ecbc_ha[] = {
     "65da73b8de06a6660ca2d36be881b708362e1f8bc12c01290aed90fdb2977667",
     0
   };
 
-    static u3j_harm _140_hex_aes_cbca_en_a[] = {{".2", u3wea_cbca_en}, {}};
     static c3_c* _140_hex_aes_cbca_en_ha[] = {
       "f85366d520b3179c5dabfb58ee1fa0554f5044f676340439db875841cd4058de",
       0
     };
-    static u3j_harm _140_hex_aes_cbca_de_a[] = {{".2", u3wea_cbca_de}, {}};
     static c3_c* _140_hex_aes_cbca_de_ha[] = {
       "8b876fbdb1849d8fbabba5e143aea0532a0e5dfff1c784d7ad15fd497ea376b1",
       0
     };
   static u3j_core _140_hex_aes_cbca_d[] =
-    { { "en", 7, _140_hex_aes_cbca_en_a, 0, _140_hex_aes_cbca_en_ha },
-      { "de", 7, _140_hex_aes_cbca_de_a, 0, _140_hex_aes_cbca_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_cbca_ha[] = {
     "420d04c03b7816b656fe4e8d9fce04f7e6d51d3d274c6511e89cfdb43ebf107e",
     0
   };
 
-    static u3j_harm _140_hex_aes_cbcb_en_a[] = {{".2", u3wea_cbcb_en}, {}};
     static c3_c* _140_hex_aes_cbcb_en_ha[] = {
       "6f961e0629c5efce47793e6a352220d355bb8fba6656c6941a68efb3ba10999d",
       0
     };
-    static u3j_harm _140_hex_aes_cbcb_de_a[] = {{".2", u3wea_cbcb_de}, {}};
     static c3_c* _140_hex_aes_cbcb_de_ha[] = {
       "7ee2f33f80612e91fda1fd84201266dea6cab596a4e23a535d1a14fb0763f1a3",
       0
     };
   static u3j_core _140_hex_aes_cbcb_d[] =
-    { { "en", 7, _140_hex_aes_cbcb_en_a, 0, _140_hex_aes_cbcb_en_ha },
-      { "de", 7, _140_hex_aes_cbcb_de_a, 0, _140_hex_aes_cbcb_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_cbcb_ha[] = {
     "1b84daab497795f2afd7238a6a6090be68b78eb6051b0ffa076b2ed9fedeebe5",
     0
   };
 
-    static u3j_harm _140_hex_aes_cbcc_en_a[] = {{".2", u3wea_cbcc_en}, {}};
     static c3_c* _140_hex_aes_cbcc_en_ha[] = {
       "b2578cf17a3095f48cc96cf7690dd7ab4f4e0b76b2578eadc1dce31a075f5b12",
       0
     };
-    static u3j_harm _140_hex_aes_cbcc_de_a[] = {{".2", u3wea_cbcc_de}, {}};
     static c3_c* _140_hex_aes_cbcc_de_ha[] = {
       "36586f89d702bedb8c2a01ea3614f61627e762488e373106cbb1b27c46e3493c",
       0
     };
   static u3j_core _140_hex_aes_cbcc_d[] =
-    { { "en", 7, _140_hex_aes_cbcc_en_a, 0, _140_hex_aes_cbcc_en_ha },
-      { "de", 7, _140_hex_aes_cbcc_de_a, 0, _140_hex_aes_cbcc_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_cbcc_ha[] = {
     "3bab1a59c7673afeb659821d54754e8e5e281243e79624fdbe4d7f85cae192c5",
     0
   };
 
-    static u3j_harm _140_hex_aes_siva_en_a[] = {{".2", u3wea_siva_en}, {}};
     static c3_c* _140_hex_aes_siva_en_ha[] = {
       "2a137039301788b8540ed81cbfafe450c9306348d02a6576f5c14f6d6f20ba81",
       0
     };
-    static u3j_harm _140_hex_aes_siva_de_a[] = {{".2", u3wea_siva_de}, {}};
     static c3_c* _140_hex_aes_siva_de_ha[] = {
       "d0130e9229e71c589429dd87843dc104bc4ee5b426400a547081d8c91a548eaa",
       0
     };
   static u3j_core _140_hex_aes_siva_d[] =
-    { { "en", 7, _140_hex_aes_siva_en_a, 0, _140_hex_aes_siva_en_ha },
-      { "de", 7, _140_hex_aes_siva_de_a, 0, _140_hex_aes_siva_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_siva_ha[] = {
     "435c5c769d2522d71ab60332bb57440d69c6803d5ca9a5faae88c825cb55d72e",
     0
   };
 
-    static u3j_harm _140_hex_aes_sivb_en_a[] = {{".2", u3wea_sivb_en}, {}};
     static c3_c* _140_hex_aes_sivb_en_ha[] = {
       "1638f56e8728f285e4175c7b514c5a4e1a24205acf33e105f0513cad7ae843cf",
       0
     };
-    static u3j_harm _140_hex_aes_sivb_de_a[] = {{".2", u3wea_sivb_de}, {}};
     static c3_c* _140_hex_aes_sivb_de_ha[] = {
       "64c9b199fffd6d31baf7457bf27d5a510121be45201b2ae5cc1d9565c0bdd0ff",
       0
     };
   static u3j_core _140_hex_aes_sivb_d[] =
-    { { "en", 7, _140_hex_aes_sivb_en_a, 0, _140_hex_aes_sivb_en_ha },
-      { "de", 7, _140_hex_aes_sivb_de_a, 0, _140_hex_aes_sivb_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_sivb_ha[] = {
     "0683f3f9067c2a16f68805c778c404179dc5df9019bbbe0f8d680a99a69e61fc",
     0
   };
 
-    static u3j_harm _140_hex_aes_sivc_en_a[] = {{".2", u3wea_sivc_en}, {}};
     static c3_c* _140_hex_aes_sivc_en_ha[] = {
       "d721486dea943efd52d9e6450f4f48dd191c89637a2f842d3ff6edfd3beecbb9",
       0
     };
-    static u3j_harm _140_hex_aes_sivc_de_a[] = {{".2", u3wea_sivc_de}, {}};
     static c3_c* _140_hex_aes_sivc_de_ha[] = {
       "3ded831b992ea100582229a4d1d9b5c80380128ae6b59b5bb36403ed13dc5d55",
       0
     };
   static u3j_core _140_hex_aes_sivc_d[] =
-    { { "en", 7, _140_hex_aes_sivc_en_a, 0, _140_hex_aes_sivc_en_ha },
-      { "de", 7, _140_hex_aes_sivc_de_a, 0, _140_hex_aes_sivc_de_ha },
-      {}
+    {
     };
   static c3_c* _140_hex_aes_sivc_ha[] = {
     "23ef582f110d28aff82b0795305b02e5a718d667bcae97091c08bc26790e7176",
@@ -270,75 +235,37 @@ static c3_c* _140_hex_lune_ha[] = {
   0
 };
 
-static u3j_harm _140_hex_coed__ed_puck_a[] = {{".2", u3wee_puck}, {}};
-static c3_c* _140_hex_coed__ed_puck_ha[] = {
-  "1bc694675842345c50b0e20a2193bb5bcbb42f163fc832431a3d1822a81e4c98",
-  0
-};
-static u3j_harm _140_hex_coed__ed_sign_a[] = {{".2", u3wee_sign}, {}};
-static c3_c* _140_hex_coed__ed_sign_ha[] = {
-  "34ad749bf8443611cbf1f7de90a066318bd12be36f2f7f6f55281f6f7ed79754",
-  0
-};
-static u3j_harm _140_hex_coed__ed_veri_a[] = {{".2", u3wee_veri}, {}};
 static c3_c* _140_hex_coed__ed_veri_ha[] = {
   "047a7eeccb2e68aeeee631b6db86e11a5a3aa9e179660553eca6304327612dcf",
   0
 };
-static u3j_harm _140_hex_coed__ed_shar_a[] = {{".2", u3wee_shar}, {}};
 static c3_c* _140_hex_coed__ed_shar_ha[] = {
   "52d3b0a2f51f2b0a9dd72bb33db38c73dc873029c365d871d0559a1472a80e72",
   0
 };
 
-  static u3j_harm _140_hex_coed__ed_point_add_a[] =
-      {{".2", u3wee_point_add}, {}};
-
-  static u3j_harm _140_hex_coed__ed_scalarmult_a[] =
-      {{".2", u3wee_scalarmult}, {}};
   static c3_c* _140_hex_coed__ed_scalarmult_ha[] = {
     "72e71cd3aa3af429cd65baa78632500c60edd1d4c82a39d3ba7f231ef97e6316",
     0
   };
 
-  static u3j_harm _140_hex_coed__ed_scalarmult_base_a[] =
-      {{".2", u3wee_scalarmult_base}, {}};
   static c3_c* _140_hex_coed__ed_scalarmult_base_ha[] = {
     "976fdb8251f9b767af689a0d2c41b88941921bf53777c1ceeb5297511021f9d8",
     0
   };
 
-  static u3j_harm _140_hex_coed__ed_add_scalarmult_scalarmult_base_a[] =
-      {{".2", u3wee_add_scalarmult_scalarmult_base}, {}};
   static c3_c* _140_hex_coed__ed_add_scalarmult_scalarmult_base_ha[] = {
     "11819071c24a2d7b36daea6e16c78b2e05f9ca3e857cf4815ffe652ce677a61a",
     0
   };
 
-  static u3j_harm _140_hex_coed__ed_add_double_scalarmult_a[] =
-      {{".2", u3wee_add_double_scalarmult}, {}};
   static c3_c* _140_hex_coed__ed_add_double_scalarmult_ha[] = {
     "0fab78a1e890e53cecade1c22b95813db77e066044e33417a0919695b6cde9ba",
     0
   };
 
 static u3j_core _140_hex_coed__ed_d[] =
-  { { "sign", 7, _140_hex_coed__ed_sign_a, 0, _140_hex_coed__ed_sign_ha },
-    { "puck", 7, _140_hex_coed__ed_puck_a, 0, _140_hex_coed__ed_puck_ha },
-    { "veri", 7, _140_hex_coed__ed_veri_a, 0, _140_hex_coed__ed_veri_ha },
-    { "shar", 7, _140_hex_coed__ed_shar_a, 0, _140_hex_coed__ed_shar_ha },
-    { "point-add", 7, _140_hex_coed__ed_point_add_a, 0, 0 },
-    { "scalarmult", 7, _140_hex_coed__ed_scalarmult_a, 0,
-      _140_hex_coed__ed_scalarmult_ha },
-    { "scalarmult-base", 7, _140_hex_coed__ed_scalarmult_base_a, 0,
-      _140_hex_coed__ed_scalarmult_base_ha },
-    { "add-scalarmult-scalarmult-base", 7,
-      _140_hex_coed__ed_add_scalarmult_scalarmult_base_a, 0,
-      _140_hex_coed__ed_add_scalarmult_scalarmult_base_ha },
-    { "add-double-scalarmult", 7,
-      _140_hex_coed__ed_add_double_scalarmult_a, 0,
-      _140_hex_coed__ed_add_double_scalarmult_ha },
-    {}
+  {
   };
 static c3_c* _140_hex_coed__ed_ha[] = {
   "7a44a962aa72933588b5c99a8b68ebac21ce3c4710c081cb66b3599b45af9ced",
@@ -368,66 +295,30 @@ static c3_c* _140_hex_hmac_ha[] = {
   0
 };
 
-  static u3j_harm _140_hex_argon2_a[] = {{".2", u3we_argon2}, {}};
   static c3_c* _140_hex_argon2_ha[] = {
     "4df7cec141ffa2cc76b058846474ca42cc9840666ee3e7e80e565803e83ea98b",
     0
   };
 static u3j_core _140_hex_argon_d[] =
-  { { "argon2", 511, _140_hex_argon2_a, 0, _140_hex_argon2_ha },
-    {}
+  {
   };
 static c3_c* _140_hex_argon_ha[] = {
   "dc704c786192ecd09d4c206a8f28db3202b6e0eb03e3ce63a95987510ac312d6",
   0
 };
 
-  static u3j_harm _140_hex_scr_pbk_a[] = {{".2", u3wes_pbk, c3y}, {}};
-  static c3_c* _140_hex_scr_pbk_ha[] = { 0 };
-  static u3j_harm _140_hex_scr_pbl_a[] = {{".2", u3wes_pbl, c3y}, {}};
-  static c3_c* _140_hex_scr_pbl_ha[] = { 0 };
-  static u3j_harm _140_hex_scr_hsh_a[] = {{".2", u3wes_hsh, c3y}, {}};
-  static c3_c* _140_hex_scr_hsh_ha[] = { 0 };
-  static u3j_harm _140_hex_scr_hsl_a[] = {{".2", u3wes_hsl, c3y}, {}};
-  static c3_c* _140_hex_scr_hsl_ha[] = { 0 };
 static u3j_core _140_hex_scr_d[] =
-  { { "pbk", 7, _140_hex_scr_pbk_a, 0, _140_hex_scr_pbk_ha },
-    { "pbl", 7, _140_hex_scr_pbl_a, 0, _140_hex_scr_pbl_ha },
-    { "hsh", 7, _140_hex_scr_hsh_a, 0, _140_hex_scr_hsh_ha },
-    { "hsl", 7, _140_hex_scr_hsl_a, 0, _140_hex_scr_hsl_ha },
-    {}
+  {
   };
 static c3_c* _140_hex_scr_ha[] = { 0 };
 
-    static u3j_harm _140_hex_secp_secp256k1_make_a[] = {{".2", u3we_make, c3y}, {}};
-    static c3_c* _140_hex_secp_secp256k1_make_ha[] = { 0 };
-    static u3j_harm _140_hex_secp_secp256k1_sign_a[] = {{".2", u3we_sign, c3y}, {}};
-    static c3_c* _140_hex_secp_secp256k1_sign_ha[] = { 0 };
-    static u3j_harm _140_hex_secp_secp256k1_reco_a[] = {{".2", u3we_reco, c3y}, {}};
-    static c3_c* _140_hex_secp_secp256k1_reco_ha[] = { 0 };
-
-      static u3j_harm _140_hex_secp_secp256k1_schnorr_sosi_a[] =
-        {{".2", u3we_sosi}, {}};
-      static c3_c* _140_hex_secp_secp256k1_schnorr_sosi_ha[] = { 0 };
-      static u3j_harm _140_hex_secp_secp256k1_schnorr_sove_a[] =
-        {{".2", u3we_sove}, {}};
-      static c3_c* _140_hex_secp_secp256k1_schnorr_sove_ha[] = { 0 };
     static u3j_core _140_hex_secp_secp256k1_schnorr_d[] =
-      { { "sosi", 7,
-          _140_hex_secp_secp256k1_schnorr_sosi_a, 0,
-          _140_hex_secp_secp256k1_schnorr_sosi_ha },
-        { "sove", 7,
-          _140_hex_secp_secp256k1_schnorr_sove_a, 0,
-          _140_hex_secp_secp256k1_schnorr_sove_ha },
-        {}
+      {
       };
     static c3_c* _140_hex_secp_secp256k1_schnorr_ha[] = { 0 };
 
   static u3j_core _140_hex_secp_secp256k1_d[] =
-    { { "make", 7, _140_hex_secp_secp256k1_make_a, 0, _140_hex_secp_secp256k1_make_ha },
-      { "sign", 7, _140_hex_secp_secp256k1_sign_a, 0, _140_hex_secp_secp256k1_sign_ha },
-      { "reco", 7, _140_hex_secp_secp256k1_reco_a, 0, _140_hex_secp_secp256k1_reco_ha },
-      { "schnorr", 7, 0,
+    { { "schnorr", 7, 0,
         _140_hex_secp_secp256k1_schnorr_d,
         _140_hex_secp_secp256k1_schnorr_ha },
       {}
@@ -445,44 +336,28 @@ static c3_c* _140_hex_secp_ha[] = {
   0
 };
 
-  static u3j_harm _140_hex_blake2b_a[] = {{".2", u3we_blake, c3y}, {}};
   static c3_c* _140_hex_blake2b_ha[] = {
     "c432216ca53b5ad2284259167952761bb1046e280268c4d3b9ca70a2024e1934",
     0
   };
 static u3j_core _140_hex_blake_d[] =
-  { { "blake2b", 7, _140_hex_blake2b_a, 0, _140_hex_blake2b_ha },
-    {}
+  {
   };
 static c3_c* _140_hex_blake_ha[] = {
   "ff30d99ffb3e13d8aa50b2b8461c8edfabf0e76de22312d16d1d6daaf3636b5f",
   0
 };
 
-  static u3j_harm _140_hex_kecc_k224_a[] =
-    {{".2", u3we_kecc224, c3y, c3y, c3y}, {}};
-  static u3j_harm _140_hex_kecc_k256_a[] =
-    {{".2", u3we_kecc256, c3y, c3y, c3y}, {}};
-  static u3j_harm _140_hex_kecc_k384_a[] =
-    {{".2", u3we_kecc384, c3y, c3y, c3y}, {}};
-  static u3j_harm _140_hex_kecc_k512_a[] =
-    {{".2", u3we_kecc512, c3y, c3y, c3y}, {}};
 static u3j_core _140_hex_kecc_d[] =
-  { { "k224", 7, _140_hex_kecc_k224_a, 0, no_hashes },
-    { "k256", 7, _140_hex_kecc_k256_a, 0, no_hashes },
-    { "k384", 7, _140_hex_kecc_k384_a, 0, no_hashes },
-    { "k512", 7, _140_hex_kecc_k512_a, 0, no_hashes },
-    {}
+  {
   };
 
-  static u3j_harm _140_hex_ripemd_160_a[] = {{".2", u3we_ripe, c3y}, {}};
   static c3_c* _140_hex_ripemd_160_ha[] = {
     "176684b29926a01f5c60fa584e4691b0cbdc9b93608dcbe7d0cf3585683fa42f",
     0
   };
 static u3j_core _140_hex_ripe_d[] =
-  { { "ripemd160", 7, _140_hex_ripemd_160_a, 0, _140_hex_ripemd_160_ha },
-    {}
+  {
   };
 static c3_c* _140_hex_ripe_ha[] = {
   "b0cb16bf206c0496bb480e5759ea1afa7dee1748b64e5243c23fddb09720ebd0",
@@ -1369,55 +1244,30 @@ static c3_c* _140_tri__rh_ha[] = {
   0
 };
 
-  static u3j_harm _140_tri__og_raw_a[] = {{".2", u3weo_raw}, {}};
   static c3_c* _140_tri__og_raw_ha[] = {
     "bbcbefc237dbebf6c141ba14fd9e0464a836127fd123d10da5f121e82d49ebdb",
     0
   };
 static u3j_core _140_tri__og_d[] =
-  { { "raw", 7, _140_tri__og_raw_a, 0, _140_tri__og_raw_ha },
-    {}
+  {
   };
 static c3_c* _140_tri__og_ha[] = {
   "74b9ae67eeabbffcff969ac7fdc7f4f0f4f67af64931e969bcac50d084e15fc0",
   0
 };
 
-  static u3j_harm _140_tri__sha_sha1_a[] = {{".2", u3we_sha1}, {}};
   static c3_c* _140_tri__sha_sha1_ha[] = {
     "75aababa0688619d9df36238269119302a64ad2e3c69c53bd0057fe6b1abaf0c",
     0
   };
 static u3j_core _140_tri__sha_d[] =
-  { { "sha1", 7, _140_tri__sha_sha1_a, 0, _140_tri__sha_sha1_ha },
-    {}
+  {
   };
 static c3_c* _140_tri__sha_ha[] = {
   "3c22d2f8719cb626e8dfe1a4206bcbc14b678c1422c48322054b40f84416d557",
   0
 };
 
-static u3j_harm _140_tri_shax_a[] = {{".2", u3we_shax}, {}};
-static c3_c* _140_tri_shax_ha[] = {
-  "0fc53de3ddc8b8f84a46136f1728fa3ed66a5113888d14907589d16bf5927ad8",
-  0
-};
-static u3j_harm _140_tri_shay_a[] = {{".2", u3we_shay}, {}};
-static c3_c* _140_tri_shay_ha[] = {
-  "b6dbc72e15c2204f83f902619b7a60328f29c9d302ddb35c435111dea28c5470",
-  0
-};
-static u3j_harm _140_tri_shas_a[] = {{".2", u3we_shas}, {}};
-static c3_c* _140_tri_shas_ha[] = {
-  "5230583767b7625b3496248ed03b6b94c1d4ee9b26342f9390bf999ec9b6cfdb",
-  0
-};
-static u3j_harm _140_tri_shal_a[] = {{".2", u3we_shal}, {}};
-static c3_c* _140_tri_shal_ha[] = {
-  "3242912e29e3e1ed8d1a395cc860a82d78961b4278ed79bbdeb37cb5615bbf20",
-  0
-};
-
   static u3j_harm _140_ob_fein_a[] = {{".2", u3we_fein_ob}, {}};
   static c3_c* _140_ob_fein_ha[] = {
     0
@@ -1452,10 +1302,6 @@ static u3j_core _140_tri_d[] =
   { "og",   7, 0, _140_tri__og_d,   _140_tri__og_ha   },
 
   { "sha",  7, 0,               _140_tri__sha_d, _140_tri__sha_ha },
-  { "shax", 7, _140_tri_shax_a, 0,               _140_tri_shax_ha },
-  { "shay", 7, _140_tri_shay_a, 0,               _140_tri_shay_ha },
-  { "shas", 7, _140_tri_shas_a, 0,               _140_tri_shas_ha },
-  { "shal", 7, _140_tri_shal_a, 0,               _140_tri_shal_ha },
 
   { "ob", 3, 0, _140_ob_d, _140_ob_ha, _140_ob_ho },
   {}
@@ -2210,10 +2056,6 @@ static u3j_core _139_tri_d[] =
   { "og",   7, 0, _140_tri__og_d,   no_hashes },
 
   { "sha",  7, 0,               _140_tri__sha_d, no_hashes },
-  { "shax", 7, _140_tri_shax_a, 0,               no_hashes },
-  { "shay", 7, _140_tri_shay_a, 0,               no_hashes },
-  { "shas", 7, _140_tri_shas_a, 0,               no_hashes },
-  { "shal", 7, _140_tri_shal_a, 0,               no_hashes },
 
   { "ob", 3, 0, _140_ob_d, no_hashes, _140_ob_ho },
   {}
@@ -2390,10 +2232,6 @@ static u3j_core _138_tri_d[] =
   { "og",   7, 0, _140_tri__og_d,   no_hashes },
 
   { "sha",  7, 0,               _140_tri__sha_d, no_hashes },
-  { "shax", 7, _140_tri_shax_a, 0,               no_hashes },
-  { "shay", 7, _140_tri_shay_a, 0,               no_hashes },
-  { "shas", 7, _140_tri_shas_a, 0,               no_hashes },
-  { "shal", 7, _140_tri_shal_a, 0,               no_hashes },
 
   { "ob", 3, 0, _140_ob_d, no_hashes, _140_ob_ho },
   {}
diff --git a/pkg/noun/manage.c b/pkg/noun/manage.c
index d880a42032..55c6b9d6c2 100644
--- a/pkg/noun/manage.c
+++ b/pkg/noun/manage.c
@@ -17,12 +17,10 @@
 #include "jets/k.h"
 #include "log.h"
 #include "nock.h"
-#include "openssl/crypto.h"
 #include "options.h"
 #include "platform/rsignal.h"
 #include "retrieve.h"
 #include "trace.h"
-#include "urcrypt.h"
 #include "vortex.h"
 #include "xtract.h"
 
@@ -1992,22 +1990,6 @@ _cm_free_ssl(void* tox_v
 
 extern void u3je_secp_init(void);
 
-/* _cm_crypto(): initialize openssl and crypto jets.
-*/
-static void
-_cm_crypto()
-{
-  /* Initialize OpenSSL with loom allocation functions. */
-  if ( 0 == CRYPTO_set_mem_functions(&_cm_malloc_ssl,
-                                     &_cm_realloc_ssl,
-                                     &_cm_free_ssl) ) {
-    u3l_log("%s", "openssl initialization failed");
-    abort();
-  }
-
-  u3je_secp_init();
-}
-
 /* _cm_realloc2(): gmp-shaped realloc.
 */
 static void*
@@ -2031,7 +2013,6 @@ u3m_init(size_t len_i)
 {
   _cm_limits();
   _cm_signals();
-  _cm_crypto();
 
   //  make sure GMP uses our malloc.
   //
