diff --git a/pkg/c3/portable.h b/pkg/c3/portable.h
index 002276834b..0995b21fcc 100644
--- a/pkg/c3/portable.h
+++ b/pkg/c3/portable.h
@@ -36,7 +36,6 @@
 #     include <sys/time.h>
 #     include <sys/resource.h>
 #     include <sys/mman.h>
-#     include <sys/sendfile.h>
 
 #   elif defined(U3_OS_osx)
 #     include <ctype.h>
@@ -145,6 +144,8 @@
 #     error "port: LoomBase"
 #   endif
 
+#define U3_OS_LoomBits 30
+#define U3_OS_LoomBase 0
 
   /** Private C "extensions."
   ***
diff --git a/pkg/noun/manage.c b/pkg/noun/manage.c
index d880a42032..e585d3403e 100644
--- a/pkg/noun/manage.c
+++ b/pkg/noun/manage.c
@@ -2029,8 +2029,6 @@ _cm_free2(void* tox_v, size_t siz_i)
 void
 u3m_init(size_t len_i)
 {
-  _cm_limits();
-  _cm_signals();
   _cm_crypto();
 
   //  make sure GMP uses our malloc.
